grails-app/services/org/pih/warehouse/invoice/InvoiceService.groovy

```diff
@@ -407,4 +409,67 @@ class InvoiceService {
+                count(order_item_id) 
+            from
+            (
+                select distinct 
+                    order_id, 
+                    order_number,
+                    order_item_id,
+                    quantity_ordered,
+                    sum(quantity_invoiced) as quantity_invoiced,
+                    sum(quantity_invoiced) >= quantity_ordered as invoiced
+                from
+                ( 
+                    select distinct
+                        `order`.id as order_id,
+                        `order`.order_number as order_number,
+                        order_item.id as order_item_id,
+                        order_item.quantity as quantity_ordered,
+                        invoice_item.quantity as quantity_invoiced
+                    from shipment_invoice
+                    join invoice_item on invoice_item.id = shipment_invoice.invoice_item_id
+                    join shipment_item on shipment_invoice.shipment_item_id = shipment_item.id
+                    join order_shipment on shipment_item.id = order_shipment.shipment_item_id
+                    join order_item on order_shipment.order_item_id = order_item.id
+                    join `order` on `order`.id = order_item.order_id
+                    join invoice on invoice.id = invoice_item.invoice_id
+                    join invoice_type on invoice_type.id = invoice.invoice_type_id
+                    where invoice.date_posted is not null and invoice_type.code <> :prepaymentTypeCode
+                    
+                    union all 
+                
+                    select distinct 
+                        `order`.id as order_id,
+                        `order`.order_number as order_number,
+                         order_item.id as order_item_id,
+                        order_item.quantity as quantity_ordered,
+                        invoice_item.quantity as quantity_invoiced\t
+                    from order_invoice
+                    join invoice_item on order_invoice.invoice_item_id = invoice_item.id
+                    join order_item on order_item.id = order_invoice.order_item_id
+                    join `order` on `order`.id = order_item.order_id
+                    join invoice on invoice.id = invoice_item.invoice_id
+                    join invoice_type on invoice_type.id = invoice.invoice_type_id
+                    where invoice.date_posted is not null and invoice_type.code <> :prepaymentTypeCode
+                ) as order_item_invoice_union
+                where order_id = :orderId
+                group by order_id, order_number, order_item_id, quantity_ordered
+                having invoiced = 1
+            ) as order_item_invoice_summary
+        """
+        SQLQuery sqlQuery = sessionFactory.currentSession.createSQLQuery(query)
+        final count = sqlQuery.with {
+            setString("orderId", order.id)
+            setString("prepaymentTypeCode", InvoiceTypeCode.PREPAYMENT_INVOICE.toString())
```

Comment:
As I mentioned in the ticket, I'm not sure this is correct. If there are requirements for excluding prepayment invoices in the "invoiced" count, then we can move this into the query (we don't need to use query parameters).

Reply:
I explained in the ticket why it should be excluded and how it is excluded now in this method: https://github.com/openboxes/openboxes/blob/c6bc349696d2a5d05a56dff5def061d02bcaff6e/grails-app/domain/org/pih/warehouse/order/OrderItem.groovy#L344-L354

I've added it as query parameters to follow the convention you did with `orderId` - I didn't wanna mix things up, to have one parameter via query parameters, and one inside the query directly.

Reply:
https://github.com/openboxes/openboxes/blob/develop/grails-app/migrations/views/order-summary-helper-views.sql#LL87C43-L87C46 This one here is doing that for OrderSummary (I know it's not intuitive and requires at least a comment), Katarzyna also linked tickets regarding the functionality in the OBPIH-5610.